File: Practice/Task4 python/37/task37.py
#!/usr/bin/python3
import requests
import sys
import os
import re
from bs4 import BeautifulSoup
from collections import Counter

def remove_tags(html):
    soup = BeautifulSoup(html, "html.parser") 
    for data in soup(['style', 'script']):
         data.decompose() 
    return ' '.join(soup.stripped_strings)

if (len(sys.argv) != 2):
	print("task37.py: missing file operands")
	print("Try './task37.py --help' for more information")

elif (sys.argv[1] == "--help"):
	print("Usage: task37.py URL or *.html")
	print("Select the 100 most used words in HTML text")

elif (sys.argv[1] == "--test"):
	os.system("./test37.bash")

elif (len(sys.argv) == 2):
	if(os.path.isfile(sys.argv[1])):
		text = open(sys.argv[1], "r", encoding='utf-8').read()
		with open('output.txt', 'w') as f:
			f.write(remove_tags(text))
	else:		
		#url = 'https://www.gnu.org/software/libc/manual/html_node/Opening-and-Closing-Files.html'
		page = requests.get(sys.argv[1])
		with open ('output.txt', 'w') as file:
			file.write(remove_tags(page.content))
	words = re.findall(r'\w+', open('output.txt').read().lower())
	count = Counter(words).most_common(100)
	print(count)

File: Practice/Task5 perl/37/task37.pl
#!/usr/bin/perl
if ($ARGV[0] eq "--help")
{
	print "Usage: task37.pl URL or *.html\n";
	print "Select the 100 most used words in HTML text\n";	
}
elsif ($ARGV[0] eq "--test")
{
	system("./test37.bash");
}
elsif (scalar @ARGV != 1)
{
	print "task37.pl: missing file operands\n";
	print "Try 'task37.pl --help' for more information\n";
}
else
{
	use strict;
	use warnings;
	use LWP::Simple qw(get);
	use HTML::Strip;
	use HTML::TokeParser::Simple;
	my $out = "output.txt";

 	if (-e $ARGV[0])
	{
	
                open FH, ">", $out;
                my $p = HTML::TokeParser::Simple->new ($ARGV[0]);
                while ( my $token = $p->get_token )
                {
                       print FH $token->as_is if $token->is_text;
                }
                close (FH);
	}
	else
	{
		my $html = get $ARGV[0];
                my $hs = HTML::Strip->new();
                my $clean = $hs->parse($html);
		$hs->eof;		
		open FH, ">", $out;
		print FH $clean;
		close (FH);
	}
	my %hash;
        open my $fh, '<', $out or die $!;
        while (<$fh>)
        {
		$hash{ lc $1 }++ while /(\w+)/g;
        }
        close $fh;
        my $i = 0;
        for my $word ( sort { $hash{$b} <=> $hash{$a} || $a cmp $b } keys %hash )
        {
		print "$i. $word: ($hash{$word})\n" if $i++ < 100 or !( $i % 1000 );
        }

}
